plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/kotlin/KSPropertyDeclarationImpl.kt

```diff
@@ -49,7 +51,9 @@ class KSPropertyDeclarationImpl private constructor(val ktProperty: KtProperty)
         if (getter != null) {
             KSPropertyGetterImpl.getCached(getter)
         } else {
-            null
+            (ResolverImpl.instance.resolveDeclaration(ktProperty) as? PropertyDescriptor)?.getter?.let {
```

Comment:
Can we have `KSPropertyGetterSyntheticImpl` to avoid resolution?

Reply:
Same for primary constructor, although which was merged already. My main concern is performance.

Reply:
I don't get the performance part. It is already lazy, do you mean to defer the accessor creation like
```
KSPropertyGetterSyntheticImpl(val ktProperty: KtProperty) {}
```
and then lazily evaluate fields inside that class? I am not sure if it is practically helpful though, since for whatever operation you perform you always need to resolve first, this will only benefit the use case where getter field is only accessed.

Reply:
Most of the information are pretty predictable and can be hard-coded. For example, it has no location, annotation and modifier. Descriptor is only needed when its parameter is asked for.

In the cases of setter and primary constructor, descriptors are probably not needed at all.